webdslstatix/trans/static-semantics/webdsl.stx

```diff
@@ -26,13 +26,20 @@ imports
 signature
 
   sorts
-    TYPE // semantic type
-    BOOL // used as return values of functional constraints
+    TYPE   // semantic type
+    BOOL   // used as return values of functional constraints
+    ORIGIN // dummy data: used to guide scope graph differ
 
   constructors
     TRUE : BOOL
     FALSE : BOOL
 
+    FUNCTION_ORIGIN: list(FormalArg) -> ORIGIN
+    NATIVE_FUNCTION_ORIGIN: list(NativeType) -> ORIGIN
+    BUILTIN_ORIGIN: list(TYPE) -> ORIGIN
+    PROP_ORIGIN: ORIGIN // prop names should be unique. TODO: how about non-unique properties in incorrect programs?
```

Comment:
I guess this TODO can be ignored. It indicates suboptimal behavior for programs like
```
entity A {
  x: Int
  x: String
}
```
where an irrelevant change is made.